packages/internal/src/debug.js

```diff
@@ -1,41 +1,108 @@
 // @jessie-check
 
+/**
+ * @import {OptPrefixPath} from './types-internal.js';
+ */
+
+// TODO This is top level static mutable state. Why did it pass at-jessie-check?
 let debugInstance = 1;
 
+const logDisabled = (..._args) => {};
+logDisabled.sub = _ => logDisabled;
+harden(logDisabled);
+
 /**
- * @param {string} name
+ * @typedef {typeof logDisabled} TraceLogger
+ */
+
+/**
+ * @param {OptPrefixPath} optPrefixPath
+ * @param {number} rootInstanceNum
  * @param {boolean | 'verbose'} enable
+ * @returns {TraceLogger}
  */
-export const makeTracer = (name, enable = true) => {
-  debugInstance += 1;
+const makeSubTracer = (optPrefixPath, rootInstanceNum, enable) => {
+  // DO NOT ASSIGN to the parameter variables above, since they are captured
+  // by closures below.
+  const makeKey = () => {
+    // As much as possible, postpone anything expensive to here, since this is
+    // only called when a tracer or sub-tracer function is called. Not when it
+    // is made.
+    let prefix;
+    // instead, make a local assignable copy
+    let optPath = optPrefixPath;
+    while (optPath !== undefined) {
+      const [step, optRestSteps] = optPath;
+      if (prefix === undefined) {
+        prefix = step;
+      } else {
+        prefix = `${step}.${prefix}`;
+      }
+      optPath = optRestSteps;
+    }
+    return `----- ${prefix}.${rootInstanceNum} `;
+  };
+  const sub = prefixStep =>
+    makeSubTracer([prefixStep, optPrefixPath], rootInstanceNum, enable);
+
+  // Because `debugCount` is pre-incremented before output, starting it at
+  // `1` means the output counts start at `2`.
```

Comment:
Reviewers, is this intentional? Is it too late to fix?

Reply:
Reviewers, 
Does anyone actually use the rendered `debugCount` for anything? When you see this free floating number, to you remember what it means?

Reply:
https://github.com/Agoric/agoric-sdk/issues/11848

Reply:
Resolving here. But note the conversation below that I won't Resolve I check this removal with the kernel and SRE teams, as suggested by @turadg .